src/main/java/org/apache/sysml/runtime/controlprogram/paramserv/spark/SparkPSWorker.java

```diff
@@ -73,5 +88,18 @@ private void configureWorker(Tuple2<Integer, Tuple2<MatrixBlock, MatrixBlock>> i
 
 		// Initialize the buffer pool and register it in the jvm shutdown hook in order to be cleanuped at the end
 		RemoteParForUtils.setupBufferPool(_workerID);
+
+		// Create the ps proxy
+		_ps = PSRpcFactory.createSparkPSProxy(_host);
```

Comment:
On configuring the worker, you might want to double check that the list objects are deserialized with the right update status. Otherwise, this might be one of the reasons of invalid cleanups again.